src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -8936,4 +8938,554 @@ public void NotExhaustive_LongList()
+              IL_0029:  stloc.0
+              IL_002a:  ldloc.0
+              IL_002b:  brfalse.s  IL_004a
+              IL_002d:  ldarg.1
+              IL_002e:  ldloc.1
+              IL_002f:  ldc.i4.1
+              IL_0030:  sub
+              IL_0031:  ldelem.ref
+              IL_0032:  isinst     "string"
+              IL_0037:  stloc.2
+              IL_0038:  ldloc.2
+              IL_0039:  brfalse.s  IL_004a
+              IL_003b:  ldloc.2
+              IL_003c:  ldstr      ")"
+              IL_0041:  call       "bool string.op_Equality(string, string)"
+              IL_0046:  brfalse.s  IL_004a
+              IL_0048:  ldloc.0
+              IL_0049:  ret
+              IL_004a:  ldnull
+              IL_004b:  ret
+            }
+            """);
+    }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/60091")]
+    [InlineData("var")]
+    [InlineData("string")]
+    public void SlicePattern_FirstDeclarationPatternOfSameType_String_CodeGen(string typeSpecifier)
+    {
+        var source = $$"""
+            class C
+            {
+                string M(string s)
+                {
+                    if (s is [..{{typeSpecifier}} start, ')'])
+                    {
+                        return start;
+                    }
+
+                    return null;
+                }
+            }
+            """;
+
+        var comp = CreateCompilationWithIndexAndRangeAndSpan(source);
+        comp.VerifyEmitDiagnostics();
+
+        CompileAndVerify(comp).VerifyIL("C.M", """
+            {
+              // Code size       42 (0x2a)
+              .maxstack  4
+              .locals init (string V_0, //start
+                            int V_1)
+              IL_0000:  ldarg.1
+              IL_0001:  brfalse.s  IL_0028
+              IL_0003:  ldarg.1
+              IL_0004:  callvirt   "int string.Length.get"
+              IL_0009:  stloc.1
+              IL_000a:  ldloc.1
+              IL_000b:  ldc.i4.1
+              IL_000c:  blt.s      IL_0028
+              IL_000e:  ldarg.1
+              IL_000f:  ldloc.1
+              IL_0010:  ldc.i4.1
+              IL_0011:  sub
+              IL_0012:  callvirt   "char string.this[int].get"
+              IL_0017:  ldc.i4.s   41
+              IL_0019:  bne.un.s   IL_0028
+              IL_001b:  ldarg.1
+              IL_001c:  ldc.i4.0
+              IL_001d:  ldloc.1
+              IL_001e:  ldc.i4.1
+              IL_001f:  sub
+              IL_0020:  callvirt   "string string.Substring(int, int)"
+              IL_0025:  stloc.0
+              IL_0026:  ldloc.0
+              IL_0027:  ret
+              IL_0028:  ldnull
+              IL_0029:  ret
+            }
+            """);
+    }
+
+    [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/60091")]
+    [InlineData("var")]
+    [InlineData("ReadOnlySpan<string>")]
+    public void SlicePattern_FirstDeclarationPatternOfSameType_ReadOnlySpan_CodeGen(string typeSpecifier)
```

Comment:
I notice tests where the simple slice is at the start or in the middle. Consider adding a test where it's at the end.

Reply:
Gentle ping on comment

Reply:
Yes, thanks, I didn't forget about it. It just so happenned that I adressed some feedback and then was busy IRL, so this stayed unfinished. And now it so happened that I'm working on my string concat PR and I want to finish it first. I'll explicitly request review from you when I return here and adress left over feedback